packages/components/src/components/hds/form/section/index.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+
+import type { ComponentLike } from '@glint/template';
+import type { HdsFormSectionHeaderSignature } from './header.ts';
+import type { HdsFormHeaderTitleSignature } from '../header/title.ts';
+import type { HdsFormHeaderDescriptionSignature } from '../header/description.ts';
+import type { HdsFormSectionMultiFieldGroupSignature } from './multi-field-group/index.ts';
+import type { HdsFormSectionMultiFieldGroupItemSignature } from './multi-field-group/item/index.ts';
+
+export interface HdsFormSectionSignature {
+  Args: {
+    hasBorder?: boolean;
+    isFullWidth?: boolean;
+  };
+  Blocks: {
+    default: [
+      {
+        Section?: ComponentLike<HdsFormSectionSignature>; // for nested sections
+        Header?: ComponentLike<HdsFormSectionHeaderSignature>;
+        HeaderTitle?: ComponentLike<HdsFormHeaderTitleSignature>;
+        HeaderDescription?: ComponentLike<HdsFormHeaderDescriptionSignature>;
+        MultiFieldGroup?: ComponentLike<HdsFormSectionMultiFieldGroupSignature>;
+        MultiFieldGroupItem?: ComponentLike<HdsFormSectionMultiFieldGroupItemSignature>;
+      },
+    ];
```

Comment:
[todo] similar to what we did above
```suggestion
import type { ComponentLike, WithBoundArgs } from '@glint/template';
import type { HdsFormSectionHeaderSignature } from './header.ts';
import type HdsFormHeaderTitleComponent from '../header/title.ts';
import type { HdsFormHeaderDescriptionSignature } from '../header/description.ts';
import type { HdsFormSectionMultiFieldGroupSignature } from './multi-field-group/index.ts';
import type { HdsFormSectionMultiFieldGroupItemSignature } from './multi-field-group/item/index.ts';

export interface HdsFormSectionSignature {
  Args: {
    hasBorder?: boolean;
    isFullWidth?: boolean;
  };
  Blocks: {
    default: [
      {
        Section?: ComponentLike<HdsFormSectionSignature>; // for nested sections
        Header?: ComponentLike<HdsFormSectionHeaderSignature>;
        HeaderTitle?: WithBoundArgs<typeof HdsFormHeaderTitleComponent, 'size'>;
        HeaderDescription?: ComponentLike<HdsFormHeaderDescriptionSignature>;
        MultiFieldGroup?: ComponentLike<HdsFormSectionMultiFieldGroupSignature>;
        MultiFieldGroupItem?: ComponentLike<HdsFormSectionMultiFieldGroupItemSignature>;
      },
    ];
```